pkg/volume/volume.go

```diff
@@ -215,20 +215,31 @@ func (v *VolumeOperation) Read() (*v1alpha1.CASVolume, error) {
 		return nil, fmt.Errorf("unable to read volume: volume name not provided")
 	}
 
-	// TODO
-	// Get the PV details & extract the SC & then SC details
-	//  Get the CAS Template name for read
-	//
-	// cas template to read a cas volume
-	castName := v.volume.Annotations[string(v1alpha1.CASTemplateKeyForVolumeRead)]
-	if len(castName) == 0 {
-		// use the DEFAULT read cas template otherwise
-		// TODO
-		//  Remove the use of defaults & make volume annotations mandatory
-		// for read operation
-		castName = string(v1alpha1.DefaultCASTemplateForJivaVolumeRead)
+	// check if sc name is already present, if not then extract it
+	scName := v.volume.Annotations[string(v1alpha1.StorageClassKey)]
+	if scName == "" {
+		// fetch the pv specification
+		pv, err := v.k8sClient.GetPV(v.volume.Name, mach_apis_meta_v1.GetOptions{})
+		if err != nil {
+			return nil, err
```

Comment:
Is error logged in caller?

Reply:
yes

Reply:
All errors are logged in the default logger.